### PR TITLE
fix: `ApproveAllowanceValidator` NPEs

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
- Cherry-pick additional fixes from #12811

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/ApproveAllowanceValidator.java [hunk 1]
```diff
@@ -129,7 +130,7 @@ private void validateFungibleTokenAllowances(
         for (final var allowance : tokenAllowances) {
             final var owner = allowance.owner();
             final var spender = allowance.spenderOrThrow();
-            final var token = tokenStore.get(allowance.tokenIdOrThrow());
+            final var token = tokenStore.get(allowance.tokenIdOrElse(TokenID.DEFAULT));
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/ApproveAllowanceValidator.java:133]: Shouldn't this fail in pureChecks ?

**Review by Neeharika-Sompalli — COMMENTED** [review]
This change LGTM. But I think a check is missing in `pureChecks`

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/ApproveAllowanceValidator.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/ApproveAllowanceValidator.java]: Agreed that is more consistent @Neeharika-Sompalli , fixed!

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thanks @tinker-michaelj

**Pull request merged**